### PR TITLE
Add CMake support

### PR DESCRIPTION
**Pull request by jakobrobert**
I have now added CMake support. I have tested it under Linux (Ubuntu) only. You can test it under MacOS, and it everything works and you have merged it I will try to get it running under Windows.

I have adjusted the README accordingly.

As well, I have fixed some bug in `test_mmap_snd.cpp`, it included `string`, but for `memcmp` the header `cstring` is required.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
+cmake_minimum_required(VERSION 3.0)
+
+set(CMAKE_C_COMPILER clang)
+set(CMAKE_CXX_COMPILER clang++)
+
+project(AudacityNoiseReduction VERSION 1.0.0)
+
+add_compile_options(-std=c++11 -Wall -Wextra -pedantic)
```
Comment (tals) [inline CMakeLists.txt:8]: a bit better to use `CXX_STANDARD`
https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html

**Comment by tals** [comment]
That's awesome!

hmm I think <string.h> is the right one :)
https://en.cppreference.com/w/c/string/byte/memcmp

**Comment by tals** [comment]
lgtm otherwise 👍👍👍

File: CMakeLists.txt (hunk 1 above)
Comment (jakobrobert) [reply CMakeLists.txt:8]: Okay, thanks. I have read about it, but I was not sure if it conflicts with add_compile_options. I assumed it kind of replaces all compiler flags.

Or does it actually add them?

This means if you have something like
```
add_compile_options(-Wall)
add_compile_options(-Wextra)
```

that they are combined?

**Comment by jakobrobert** [comment]
> 
> 
> That's awesome!
> 
> hmm I think <string.h> is the right one :)
> https://en.cppreference.com/w/c/string/byte/memcmp

Okay, fine. I never really understood the difference between the xxx.h and cxxx headers :D But I thought the cxxx heades are more c++ style.

File: CMakeLists.txt (hunk 1 above)
Comment (tals) [reply CMakeLists.txt:8]: yeah but its not a big deal either way :)

**Pull request merged**